src/components/molecules/ExecutionTableRow/ExecutionTableRow.tsx

```diff
@@ -2,29 +2,63 @@ import React, {useContext} from 'react';
 
 import {ExecutionStepIcon} from '@molecules';
 
-import {DashboardContext} from '@organisms/DashboardContainer/DashboardContainer';
+import {DashboardInfoPanelContext} from '@contexts';
 
+import ExecutionDuration from './ExecutionDuration';
 import {ExecutionName} from './ExecutionName';
 import {ExecutionSequenceNumber} from './ExecutionSequenceNumber';
-import {ExecutionStartEndTime} from './ExecutionStartEndTime';
+import ExecutionStartDateTime from './ExecutionStartDateTime';
 import {StyledChevronRightIcon, StyledSpace} from './ExecutionTableRow.styled';
 
+const leftPadding = 130;
+const rightPadding = 94;
+
+const gapWidth = 15;
+const durationTimeWidth = 60;
+
 const ExecutionTableRow: React.FC<any> = props => {
-  const {isSecondLevelOpen} = useContext(DashboardContext);
+  const {
+    size: {width},
+  } = useContext(DashboardInfoPanelContext);
 
   const {status, startTime, name, endTime, duration, index} = props;
 
+  const spaceLeft = width ? width - leftPadding - rightPadding : 0;
+
+  const isDurationTextVisible = spaceLeft > 600;
```

Comment:
A lot of magic numbers. If you can specify what each magic number means - do it. If no - leave it as is

Reply:
done